src/mqueryfront/src/QueryResultsStatus.js

```diff
@@ -44,35 +41,53 @@ function MatchItem(props) {
         ));
     }
 
-    const download_url =
-        API_URL +
-        "/download?job_id=" +
-        encodeURIComponent(props.qhash) +
-        "&ordinal=" +
-        encodeURIComponent(props.ordinal) +
-        "&file_path=" +
-        encodeURIComponent(props.file);
-
-    const path = require("path");
-
     return (
         <tr>
             <td>
                 <div className="row m-0 text-truncate">
                     <div className="text-truncate" style={{ minWidth: 50 }}>
-                        <a
-                            href={download_url}
-                            data-toggle="tooltip"
-                            title={props.file}
-                        >
+                        <div>
+                            {props.meta.sha256.display_text}
+                            <small>
+                                <a
+                                    href={download_url}
+                                    data-toggle="tooltip"
+                                    title={props.file}
+                                    className="text-secondary"
+                                >
+                                    <i className="fa fa-download fa-xm ml-2" />
+                                </a>
+                                <CopyToClipboard
+                                    text={props.meta.sha256.display_text}
+                                >
+                                    <span
+                                        data-toggle="tooltip"
+                                        title="Copy sha256 to clipboard"
+                                        className="hash-item"
+                                    >
+                                        <i className="fa fa-copy fa-xm ml-2" />
+                                    </span>
+                                </CopyToClipboard>
+                            </small>
+                        </div>
+                        <small class="text-secondary">
                             {path.basename(props.file)}
-                        </a>
+                            <CopyToClipboard text={path.basename(props.file)}>
+                                <span
+                                    data-toggle="tooltip"
+                                    title="Copy file name to clipboard"
```

Comment:
Again, I *think* an article is missing here

```suggestion
                                    title="Copy file name to the clipboard"
```

Reply:
Nope.